stream/stream_cb.c

```diff
@@ -0,0 +1,150 @@
+#include "config.h"
+
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include "osdep/io.h"
+
+#include "common/common.h"
+#include "common/msg.h"
+#include "common/global.h"
+#include "stream.h"
+#include "options/m_option.h"
+#include "options/path.h"
+#include "player/client.h"
+#include "libmpv/stream_cb.h"
+
+struct mpv_stream_cb_context {
+    void *user_data;
+    mpv_stream_cb_open_fn open_fn;
+    mpv_stream_cb_read_fn read_fn;
+    mpv_stream_cb_seek_fn seek_fn;
+    mpv_stream_cb_size_fn size_fn;
+    mpv_stream_cb_close_fn close_fn;
```

Comment:
So, in theory, the API user could call the setters for these fields at any time, even while custom streams are instantiated, so I think there needs to be some synchronization or some concept how to handle this.

Random idea: wrap each setter into a mutex, and also disallow setting these fields while custom streams are instantiated (by keeping a refcount) - this would make a nice API which would make it relatively hard for the API user to hurt himself.